webview/src/setup/hooks/useIsSectionComplete.ts

```diff
@@ -0,0 +1,25 @@
+import { useEffect, useState } from 'react'
+import { usePrevious } from './usePrevious'
+
+export const useIsSectionComplete = (
```

Comment:
Is there a simpler way to do this?

Reply:
Do we really need `useEffect` (and three of them)? Can't we move the state into `<Dvc>`?

If you set the initial state of `const [isComplete, setIsComplete] = useState<boolean | undefined | null>(null)` then you should know when it is updated because it should only ever be `null` when it is unset. Should then be able to use something like `if (previousIsComplete === null || isComplete === null) { return } `?

Reply:
Also if we are going to introduce `hasReceivedMessageFromVsCode` we should introduce some loading state as well.

Reply:
Here's what I've got so far: 

```
  const [isComplete, setIsComplete] = useState<boolean | null>(null)
  const previousIsComplete = usePrevious(isComplete)

  useEffect(() => {
    const isSetup = projectInitialized && !!cliCompatible

    if (hasReceivedMessageFromVsCode) {
      setIsComplete(isSetup)
    }

    if (previousIsComplete === false && isComplete) {
      setSectionCollapsed({
        [SetupSection.DVC]: true,
        [SetupSection.EXPERIMENTS]: false,
        [SetupSection.STUDIO]: false
      })
    }
  }, [
    projectInitialized,
    cliCompatible,
    isComplete,
    previousIsComplete,
    setSectionCollapsed,
    hasReceivedMessageFromVsCode
  ])
```

Is this what you had in mind?